### PR TITLE
Opt out user office from missed reservation notifications

### PR DESCRIPTION
**Pull request by r-xyz**
Hi @rptmat57 
I am opening this PR for an opt-out customization to remove User Office email from some notifications.

In first place I was thinking about removing it from the missed reservation notification. These emails already reach the abuse mailing list, same as "unauthorized tool access email" and "Out of time email"; the latter two notifications are not sent to user office, instead.
I was wondering if the discrepancy above was intended. If so, I would proceed to remove it through a Customization option, for the sake to avoid breaking changes.

I was also wondering if there are any similar cases which might make sense to include in the PR, either here or in NEMO-CE, to have the same approach.

On top of that, any other suggestion of different approaches is always welcome.

### CODE REVIEW

**Comment by rptmat57** [comment]
that's a good idea. most of the notification emails being cc'd to one or the other email addresses are purely based on the CNST NanoFab requirements at the time.
It would be a good idea to make those more flexible.

I could see the "Email addresses" part of Customizations being extended to have checkboxes to allow facilities to decide which of the emails should they be included to. This could expand to the other email addresses, and potentially also regarding the emails being sent from those email addresses.

In short:

1. Expanding Customization -> Email addresses to include checkboxes for emails they are cc'd on (not limited to User Office but this could be split into multiple PRs)
2. Expanding it further to include checkboxes for emails being sent as them. If unchecked we would leave it blank which will use the default server email from settings.py (this could also be done in a separate PR)

How does that sound?

**Comment by r-xyz** [comment]
Thanks for the feedback.
It sounds good.
Would it be something like this, but for multiple notification types?
image

I see the other way around (having subsections for addresses instead of notification type) as more complicated to implement in UI, since you can only have one sender per notification.

**Comment by rptmat57** [comment]
I am debating actually. I was originally thinking simply listing the emails sent from any of those addresses (only having choice between server email and that email) and whether they should be cced on stuff, but now I am rethinking this.

Here is what I am now thinking:

1. adding more variable to the `TemplatesCustomization` class like this:
```python
    def __init__(self, *args, **kwargs):
        for email_template, extension in self.files:
            self.__class__.variables[f"{email_template}_subject"] = ""
            self.__class__.variables[f"{email_template}_send_from"] = ""
            self.__class__.variables[f"{email_template}_cc_addresses"] = ""
        super().__init__(*args, **kwargs)
```
2. the `subject` would be a django template string, allowing things like `{{ variable }}` following the same provided vars as the ones in the email template
3. the `send_from` would have a datalist html widget with a few predefined choices (user_office, abuse, etc) and allow custom entry.
4. the `cc_addresses` would allow multiple entries
5. for `#3` and `#4` we could also simply use variable resolution to allow `{{ user_office_email }}` for example.
6. for all the default values, I think the only way here would be to create a data migration that would add all the current `send_from` and `subject` and `cc_addresses`

This is a pretty big feature, and I probably haven't thought all of it through so we can discuss more, and if that's too much I also totally get it.

Let me know your thoughts.

Thanks,
Mathieu

**Comment by r-xyz** [comment]
Hi Mathieu,
Sorry I completely lost track on this one.
Your idea seems more flexible.
If I get it correctly, in the settings it should show as one section/row for kind of notification, where you can select `send_from`, `subject`, `cc_addresses.
Brainstorm idea: would it make sense to have it in the "File & Email templates" instead? Or at least have the templates page render (read only) some of these settings; currently some templates do not indicate the `cc_addresses` nor the subject, which might be useful when setting up or making changes to the system. 

Indeed, I was going to suggest adding to the manual a Table comparison of features and notification received for Primary Owner, Backup Owners and Superusers (and maybe a similar one for user permissions, like staff, Facility, etc); I know the info is already there, both in the help text and the manual, but I personally find a bit hard to keep an overview, especially regarding notifications. Anyway, with your proposed changes, this might become overkill.

**Comment by r-xyz** [comment]
PS:
I see you started `7.0.0.dev`, should new features be based against that? Is there any 6.1.0 release expected? We are looking forward to make some of the new contributed features/improvements available to user. Please let me know if there is any way we could help with the documentation, which you also have to update on each release.

**Comment by r-xyz** [comment]
Hi @rptmat57 ,
Thanks for the release! :)
I think this got closed by mistake, nonetheless I would probably start it over if we are going to implement your new idea.
